vmngclient/utils/response.py

```diff
@@ -21,30 +21,68 @@ def __str__(self):
 
 @dataclass
 class VManageResponseDebugInfo:
-    request: Dict[str, Any]
-    response: Dict[str, Any]
-
-
-def response_debug(response: Response, headers: bool = True) -> str:
-    request_body = response.request.body
-    if isinstance(request_body, bytes) and request_body.isascii():
-        request_body = str(request_body, encoding="utf-8")
-    info = VManageResponseDebugInfo(
-        request={
-            "method": response.request.method,
-            "url": response.request.url,
-            "body": request_body,
-        },
-        response={"status": response.status_code, "reason": response.reason},
-    )
-    if len(response.text) <= 1024:
-        info.response.update({"text": response.text})
+    request: Optional[Dict[str, Any]] = None
+    response: Optional[Dict[str, Any]] = None
+
+
+def response_debug(response: Optional[Response], request: Union[Request, PreparedRequest, None]) -> str:
+    """Returns human readable string containing Request-Response contents (helpful for debugging).
+
+    Args:
+        response: Response object to be debugged (note it contains an PreparedRequest object already)
+        request: optional Request object to be debugged
+
+    Returns:
+        str
+    """
+    if request is None and response is None:
+        return ""
+    if request is None and response is not None:
+        request = response.request
+        request_body = request.body
     else:
-        info.response.update({"text(trimmed)": response.text[:128]})
-    if headers:
-        info.request.update({"headers": dict(response.request.headers.items())})
-        info.response.update({"headers": dict(response.headers.items())})
-    return pformat(dict(info.__dict__.items()), width=80, indent=0)
+        request_body = None
+    info = VManageResponseDebugInfo()
+    if request is not None:
+        info.request = {
+            "method": request.method,
+            "url": request.url,
+            "body": request_body,
+            "headers": dict(request.headers.items()),
+        }
+    if response is not None:
+        info.response = {
+            "status": response.status_code,
+            "reason": response.reason,
+            "headers": dict(response.headers.items()),
+        }
+        try:
+            json = response.json()
+            json.pop("header", None)
+            info.response.update({"json": json})
+        except JSONDecodeError:
+            if len(response.text) <= 1024:
+                info.response.update({"text": response.text})
+            else:
+                info.response.update({"text(trimmed)": response.text[:128]})
+    return pformat(dict(info.__dict__.items()), width=80, sort_dicts=False)
```

Comment:
Too many `if` keywords.
`request` variable is overwritten. Can we implement it in a more functional way?